modules/apps/object/object-test/src/testFunctional/tests/ObjectCustomValidation.testcase

```diff
@@ -2289,6 +2212,96 @@ definition {
 		ObjectPortlet.assertEntryNotPresent(entry = "Decline Entry");
 	}
 
+	@description = "LPS-144474 - Verify that, when updating an existing validation, this validation must only affect the entries added after the validation update"
+	@priority = 4
+	test CanUpdateValidation {
+		property custom.properties = "feature.flag.LPS-187846=true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("Given a custom object definition with a text field") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 192647",
+				objectName = "CustomObject192647",
+				pluralLabelName = "Custom Objects 192647");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Text",
+				fieldName = "text",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject192647");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject192647");
+		}
+
+		task ("and Given a expression builder validation set to full validation") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 192647");
+
+			ObjectAdmin.goToValidationsTab();
+
+			ObjectCustomValidation.addObjectValidation(
+				validationName = "Custom Validation",
+				validationType = "Expression Builder");
+
+			ObjectCustomValidation.gotoCustomValidation(validationName = "Custom Validation");
+
+			ObjectCustomValidation.activeValidation();
+
+			ObjectCustomValidation.gotoTab(tabName = "Conditions");
+
+			ObjectCustomValidation.insertScriptOnValidation(scriptValidation = "NOT(contains(text, 'lazy'))");
+
+			ObjectCustomValidation.addErrorMessage(errorMessage = "This entry is not valid for text entry.");
+
+			PortletEntry.save();
+		}
+
+		task ("and Given a valid object entry") {
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "text",
+				objectName = "CustomObject192647",
+				value = "Quick brown fox");
+		}
+
+		task ("When editing the custom validation") {
+			ObjectCustomValidation.gotoCustomValidation(validationName = "Custom Validation");
+
+			ObjectCustomValidation.editLabelOnCustomValidation(editValidationName = "Custom Validation Updated");
+
+			ObjectCustomValidation.gotoTab(tabName = "Conditions");
+
+			ObjectCustomValidation.insertScriptOnValidation(scriptValidation = "NOT(contains(text, 'dog'))");
+
+			ObjectCustomValidation.addErrorMessage(errorMessage = "This entry is not valid for text entry, the field contains the word: dog.");
+
+			PortletEntry.save();
+		}
+
+		task ("and When attempting to add an object entry with an invalid object entry") {
```

Comment:
When the user changes the validation logic and attempts to create an invalid object entry